### DISCUSSION TITLE
Hide failed tasks alongside completed ones

### DISCUSSION THREAD

**Discussion by lukaszlew** [iSoron/uhabits#1049] [Feature Requests] [answered]
I'm using the app as recommended by [Atomic habits](https://www.goodreads.com/book/show/40121378-atomic-habits).
I'm using the main screen as a *cue* that is starting the habit execution. 
This list is big (66 items). I organize it to go from top to bottom. Sometimes I fail a task like 'meditate before work' and I can use new option to mark it as failed [X insead of ?]. But I have no way of hiding these from the list. 

Can we add an option of hiding X from the main list?

A secondary reason is that sometimes I misclick failed task [X], marking it 'done'.
It immediately disappears from the list. I can't even find what it was.

**Comment by hiqua** [comment]
If you're using v2 (still a beta), have a look at "sort by status". I know that's only a partial solution, so let us know what you think is still missing once you use this sort.

**Reply by lukaszlew** [reply]
My sort order is carefully crafted to mirror my day. I can't afford to resort. 
Sometimes I am forced to do things out of order, but the next day I want to keep my order.
Also sorting would not hide Xs, so I can still misclick.
Perhaps Im missing something, why do you think that sorting is helpful?

**Reply by hiqua** [reply]
If you're concerned about misclicks, I'd recommend to use the long-press option.

The reason I suggested this sort is so that the top of your list is always relevant, i.e. tasks for which you haven't either failed or succeeded yet. But if you rely on the manual sort, it's indeed not helpful.

I think it's a valid and simple feature request, and it should be easy to implement, I'll just wait for the maintainer's feedback for now.

**Reply by hiqua** [reply]
We could have `Hide Completed` also hide failed habits, would that be fine for you?

**Reply by lukaszlew** [reply]
That would work for me personally.

**Reply by hiqua** [reply]
Ok I've edit the title to reflect that, hoping that others who would be unhappy with this change could chime in.

**Reply by lukaszlew** [reply]
Thanks!

**Comment by iSoron** [comment, marked as answer]
This has been implemented in the `dev` branch and should become available in Loop 2.1.

**Reply by lukaszlew** [reply]
Big Thanks!

**Reply by NomarCub** [reply]
I've tried the dev version. How can one completely uncheck a habit, to not hide it this way?
Now once I uncheck a habit, it can't go back to the `?` state. This way I can't unhide something I accidentally checked (even if I uncheck it after).

**Reply by lukaszlew** [reply]
Perhaps '?' should be treated as any other repetition (habit marking), with a lack of any repetition being treated as '?'. 
UI then could support choosing it. Perhaps with one of the options described in [this thread](https://github.com/iSoron/uhabits/discussions/1053)

**Comment by jt007** [comment]
I see this feature request #1049 is not included in the Loop 2.1 Milestone, either as open or closed. It looks like that was the intention by #iSoron by August 6.  Have I misunderstood anything?

https://github.com/iSoron/uhabits/milestone/13

I really look forward to these feature, which is important to de-clutter the list on days with several failed habits.
